### PR TITLE
Add Twenty Shared & Fix profile image rendering

### LINKED ISSUE #8601: Profile Picture - File not found in Settings
# Current behavior

{"error":"File not found"} when connected with Google

Image

### PR DESCRIPTION
**Pull request by mdrazak2001**
PR Summary: 

1. Added `Twenty Shared` Package to centralize utilitiies as mentioned in #8942 
2. Optimization of `getImageAbsoluteURI.ts` to handle edge cases

image

### CODE REVIEW

**Comment by FelixMalfait** [comment]
cc @AMoreaux fyi

Thank you @mdrazak2001! 
I think it might be a good opportunity to create a `twenty-shared`... It's a mess to have 3 times the same function 😅

Do you think you could take a shot at it today? Or should we take over on your PR? 
Thanks a lot

**Comment by AMoreaux** [comment]
Hello 👋
 
Thanks for the contribution.

Before adding the fix can you add a test including the URL pattern that broke the function? 

In this file:
https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/utils/image/__tests__/getImageAbsoluteURI.test.ts 


Moreover, since you're on this file. Could you add this test? It validate the fact we manage the risk of double slash.

```
  it('should return fully formed url if imageUrl is a relative url and start with a /', () => {
    const imageUrl = '/XXX';
    const result = getImageAbsoluteURI(imageUrl);
    expect(result).toBe('http://localhost:3000/files/XXX');
  });
```

The URL API allows you to validate these kinds of use cases. You don't need to manage the double // or the validation of the URI yourself.

If you need help or have a lack of time feel free to tell me ;)

**Comment by mdrazak2001** [comment]
> cc @AMoreaux fyi
> 
> Thank you @mdrazak2001! I think it might be a good opportunity to create a `twenty-shared`... It's a mess to have 3 times the same function 😅
> 
> Do you think you could take a shot at it today? Or should we take over on your PR? Thanks a lot

Great suggestion! I'll work on creating a `twenty-shared` utility to consolidate this function across the codebase.
As part of consolidating this utility into a shared implementation, would you like me to replace all occurrences of `getImageAbsoluteURI` across the codebase and remove the duplicate implementations in this PR itself?

**Comment by FelixMalfait** [comment]
@mdrazak2001 yes that'd be great. It might be a bit difficult to do a good configuration so let us know if you're stuck!

**Comment by mdrazak2001** [comment]
> Hello 👋
> 
> Thanks for the contribution.
> 
> Before adding the fix can you add a test including the URL pattern that broke the function?
> 
> In this file: https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/utils/image/__tests__/getImageAbsoluteURI.test.ts
> 
> Moreover, since you're on this file. Could you add this test? It validate the fact we manage the risk of double slash.
> 
> ```ts
>   it('should return fully formed url if imageUrl is a relative url and start with a /', () => {
>     const imageUrl = '/XXX';
>     const result = getImageAbsoluteURI(imageUrl);
>     expect(result).toBe('http://localhost:3000/files/XXX');
>   });
> ```
> 
> The URL API allows you to validate these kinds of use cases. You don't need to manage the double // or the validation of the URI yourself.
> 
> If you need help or have a lack of time feel free to tell me ;)


@AMoreaux:
Thank you for the detailed review!

1. I will add a test case for the specific URL pattern (?) that initially caused the bug to ensure the issue is addressed.
2. I’ll also include the additional test case you suggested, to validate proper handling of relative Image/File URLs starting with a /.

Solution Overview: I’ve realized that the root cause was related to the improper use of URL.toString() (Manual manipulating pathname casued improper encoding of  `?` to `%3F`). I’ve found a solution that leverages the URL API (Using base URL resolution) itself to handle all edge cases (e.g., leading slashes, query parameters, prevents unnecessary encoding) cleanly.

**Comment by mdrazak2001** [comment]
Hi @FelixMalfait @AMoreaux 
I’ve committed the initial setup and created the centralized twenty-shared library as suggested.
Please review it at your convenience and let me know if you have any concerns or feedback.
Thanks!

File: packages/twenty-shared/src/utils/image/__tests__/getImageAbsoluteURI.test.ts
Comment (AMoreaux) [inline packages/twenty-shared/src/utils/image/__tests__/getImageAbsoluteURI.test.ts:3]: 💯😍

File: packages/twenty-shared/src/utils/image/getImageAbsoluteURI.ts
Comment (AMoreaux) [inline packages/twenty-shared/src/utils/image/getImageAbsoluteURI.ts:3]: We talked internally, and we would like to remove this type. It will impact the function's return type.
I planned to do it myself but if you want to do it feel free :)

File: tsconfig.base.json
Comment (AMoreaux) [inline tsconfig.base.json:21]:
```suggestion
      "twenty-shared": ["packages/twenty-shared/src/index.ts"]
```
Please target the barrel file here `packages/twenty-shared/src/index.ts` instead of targeting dist

File: packages/twenty-ui/tsconfig.json
Comment (AMoreaux) [inline packages/twenty-ui/tsconfig.json:14]:
``` suggestion
"twenty-shared": ["packages/twenty-shared/src/*"]
```
Please target the barrel file here `packages/twenty-shared/src/index.ts` instead of targeting dist

File: packages/twenty-emails/tsconfig.json
Comment (AMoreaux) [inline packages/twenty-emails/tsconfig.json:11]:
```suggestion
      "twenty-shared": ["packages/twenty-shared/src/*"]
```

File: packages/twenty-front/tsconfig.json
Comment (AMoreaux) [inline packages/twenty-front/tsconfig.json:28]:
```suggestion
      "twenty-shared": ["packages/twenty-shared/src/index.ts"]
```

**Review by AMoreaux — CHANGES_REQUESTED** [review]
Lots of works, thanks 🙏

Few comments. TLDR:
- Use the barrel file
- Remove all reference to `react` or `tsx` in `twenty-shared`
- If you want you can remove the type in `getImageAbsoluteURI.ts`

I will test it in the browser asap.

**Review by AMoreaux — APPROVED** [review]
Perfect 💯

Thanks a lot for your contribution. I am waiting for the approval of @FelixMalfait for the next steps.

**Comment by charlesBochet** [comment]
@mdrazak2001 thank you! I'll merge this one today. I have unfortunately merged another fix in the meantime but your PR is bringing more, so I'll rebase it on main and merge it :)

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**
